### PR TITLE
Add additional headers to the CoreCLR package to describe its interface.

### PR DESCRIPTION
**Pull request by russellhadley**
Adds headers:
- opcode.def and openum.h to describe the MSIL opcodes.
- StaticContract.h to describe the shared contracts used in the interface between CoreCLR and JITs.
- switches.h to describe the feature defines of the CoreCLR.

Run debug builds and testruns for sanity, but changes should just impact development package contents.

@mmitche, @jkotas PTAL

### CODE REVIEW

**Comment by mmitche** [comment]
@russellhadley LGTM, please sign the CLA first before merging.

File: src/.nuget/Microsoft.DotNet.CoreCLR.Debug.Development.nuspec [hunk 1]
```diff
@@ -39,5 +39,9 @@
     <file src="..\inc\corhdr.h" target="inc\corhdr.h" />
     <file src="..\inc\corinfo.h" target="inc\corinfo.h" />
     <file src="..\inc\corjit.h" target="inc\corjit.h" />
+    <file src="..\inc\opcode.def" target="inc\opcode.def" />
+    <file src="..\inc\openum.h" target="inc\openum.h" />
+    <file src="..\inc\StaticContract.h" target="inc\StaticContract.h" />
+    <file src="..\inc\switches.h" target="inc\switches.h" />
   </files>
```
Comment (jkotas) [inline src/.nuget/Microsoft.DotNet.CoreCLR.Debug.Development.nuspec]: What are the StaticContract.h and switches.h needed for? They should not be needed to describe the CoreCLR JIT interface. (Also, StaticContract.h needs to be in all lowercase to not break Unix build.)

**Comment by janhenke** [comment]
@russellhadley I think you forgot to commit the actual header files. The build fails, because the CMake files are now referencing non-existing header files.

**Comment by russellhadley** [comment]
@jkotas, removed the StaticContract.h and switches.h from the package set.  I removed (unneeded) code on our end to get rid of the dependency.

**Comment by russellhadley** [comment]
@janhenke, the headers are out of src\inc.  I just added the code to get them copied to the package.  I'll have to take a look.

**Comment by russellhadley** [comment]
Ahh.  @janhenke, it was the casing of the header on linux that caused the problem.  Now that this header has been removed it should copy clean.

**Comment by jkotas** [comment]
Thanks. Could you please squash the two commits into one?

**Comment by russellhadley** [comment]
I just pushed a rebase but something is really wrong.  Let me straighten it.

**Comment by russellhadley** [comment]
Fixed rebase, now showing only the single commit.  I'll merge one the build comes back green.

**Pull request merged**